cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -1471,6 +1473,93 @@ class TestAzureFileSystem : public ::testing::Test {
     arrow::fs::AssertFileInfo(fs(), data.Path("dir/file0"), FileType::File);
   }
 
+  void AssertObjectContents(AzureFileSystem* fs, std::string_view path,
+                            std::string_view expected) {
+    ASSERT_OK_AND_ASSIGN(auto input, fs->OpenInputStream(std::string{path}));
+    std::string contents;
+    std::shared_ptr<Buffer> buffer;
+    do {
+      ASSERT_OK_AND_ASSIGN(buffer, input->Read(128 * 1024));
+      ASSERT_TRUE(buffer);
```

Comment:
Why do we need this?

Reply:
I guess it asserts that reading the input in fact worked (because that is also using the underlying Azure file system implementation). But we can also get rid of it, I don't have a strong opinion on it but I also don't see any harm.